eclair-core/src/main/scala/fr/acinq/eclair/db/pg/PgPaymentsDb.scala

```diff
@@ -401,5 +398,32 @@ class PgPaymentsDb(implicit ds: DataSource, lock: PgLock) extends PaymentsDb wit
     }
   }
 
+  override def removeIncomingPayment(paymentHash: ByteVector32): Unit = withMetrics("payments/remove-incoming", DbBackends.Postgres) {
+    withLock { pg =>
+      getIncomingPaymentInternal(pg, paymentHash) match {
+        case Some(incomingPayment) =>
+          incomingPayment.status match {
+            case _: IncomingPaymentStatus.Received =>
+              throw new IllegalArgumentException("Cannot remove a received incoming payment")
```

Comment:
Please don't throw. Scala is full of better ways of signaling non-exceptional errors (this case is really not an exceptional error - the only places in this file where we throw correspond to what would be an internal bug).

Just change the signature of `removeIncomingPayment` to return a `Try[Unit]` instead.